### PR TITLE
Build db file

### PR DESCRIPTION
**Pull request by sam96B**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "init": "npm i && npm run prepare && cd client && npm i",
     "build": "tsc",
+    "buildDB": "ts-node src/db/build.ts",
```
Comment (AhmedSafi97) [inline package.json:9]: `build-db` we are following this convention in naming our scripts

File: src/controllers/ProductController.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import { Request, Response, NextFunction } from 'express';
+import { sequelize } from '../db/connection';
+import { QueryTypes } from 'sequelize';
```
Comment (AhmedSafi97) [inline src/controllers/ProductController.ts:3]: third party imports then your modules

File: src/controllers/ProductController.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
```
Comment (AhmedSafi97) [inline src/controllers/ProductController.ts:4]: no empty lines between imports(imports is one block)

File: src/controllers/ProductController.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import GenericError from '../helpers/GenericError';
+import ProductQuery from '../queries/UserQuery';
+
+export default class ProductController {
```
Comment (AhmedSafi97) [inline src/controllers/ProductController.ts:8]: make sure to pull the main branch and resolve any conflict

File: src/controllers/ProductController.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  static getProducts = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+    try {
+      const products = await sequelize.query(
+        `select p.id,p.price,p.title,p.icon,p.description,coalesce(((select SUM(t.quantity) from "transactionProducts" as t join "Transactions" as ts on ts.id = t."TransactionId" where t.status = 'closed' and ts.type = 'purchase' and t."ProductId" = p.id)-(select
```
Comment (AhmedSafi97) [inline src/controllers/ProductController.ts:46]: we should have some formatting for the query

File: src/controllers/ProductController.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            join "Transactions" as ts on ts.id = t."TransactionId"
+          where
+            t.status = 'closed'
+            and ts.type = 'sale'
+            and t."ProductId" = p.id)),0) as "inStock" from "Products" as p;`,
+        {
+          type: QueryTypes.SELECT
+        }
+      );
+      res.json(products);
```
Comment (AhmedSafi97) [inline src/controllers/ProductController.ts:59]: response should be 
```js
{
    statusCode: 200,
    totalCount: the total count of the products matching the query,
    items: [... the products]
}
```

File: src/models/TransactionProductModel.ts [hunk 7]
```diff
@@ -10,11 +15,22 @@ TransactionProduct.init(
       type: DataTypes.INTEGER,
       autoIncrement: true,
       primaryKey: true
+    },
+    status: {
+      type: DataTypes.ENUM,
+      values: ['pending', 'reversed', 'closed'],
```
Comment (AhmedSafi97) [inline src/models/TransactionProductModel.ts:21]: values should be coming from the defined `TransactionType` enum, no raw string should be used here

File: src/models/TransactionProductModel.ts [hunk 8]
```diff
@@ -10,11 +15,22 @@ TransactionProduct.init(
...
+      type: DataTypes.INTEGER
+    },
+    unitPrice: {
+      type: DataTypes.INTEGER
     }
   },
   {
-    freezeTableName: true,
-    modelName: 'transaction_product',
+    modelName: 'transactionProduct',
```
Comment (AhmedSafi97) [inline src/models/TransactionProductModel.ts:33]: use PascalCase for the model name

File: src/routes/ProductsRoute.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import { Router } from 'express';
+import { getProducts } from '../controllers/ProductController';
+
+const router: Router = Router();
```
Comment (AhmedSafi97) [inline src/routes/ProductsRoute.ts:4]: router type can be inferred from the return value of `Roiuter` no need to declare it and please leave an empty line

File: src/routes/ProductsRoute.ts [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+router.get('/products', getProducts);
```
Comment (AhmedSafi97) [inline src/routes/ProductsRoute.ts:5]: the prefix `/products` should be declared at the `index.ts`

File: src/routes/index.ts [hunk 11]
```diff
@@ -2,11 +2,13 @@ import { NextFunction, Request, Response, Router } from 'express';
 import GenericError from '../helpers/GenericError';
 import authRouter from './AuthRoute';
 import userRouter from './UserRoute';
+import productsRouter from './ProductsRoute';
 
 const router = Router();
 
 router.use(authRouter);
 router.use(userRouter);
+router.use(productsRouter);
```
Comment (AhmedSafi97) [inline src/routes/index.ts:11]: where are the routers prefixes !!

**Review by AhmedSafi97 — CHANGES_REQUESTED** [review]

**Comment by AhmedSafi97** [comment]
Good job, but please for next times don't mix two tasks within the same PR

**Comment by sam96B** [comment]
> Good job, but please for next times don't mix two tasks within the same PR

noted boss

**Pull request closed without merging**